code/modules/research/chem_consumer.dm

```diff
@@ -0,0 +1,134 @@
+// 100 research points for 1 unit of the output chem. Seems balanced considering each node is 2500 points
+// An entire beaker of something can speed unlock 4 entire nodes
+#define RESEARCHPOINTS_PER_UNIT 100
+
+/obj/machinery/research_chem_consumer
+	name = "research chemical consumer"
+	desc = "Consumes chemicals to gain research points"
+	icon_state = "chemical_idle"
+	icon = 'icons/obj/machines/research_consumers.dmi'
+	anchored = TRUE
+	density = TRUE
+	var/obj/item/reagent_containers/beaker
+
+/obj/machinery/research_chem_consumer/Initialize(mapload)
+	. = ..()
+	component_parts = list()
+	component_parts += new /obj/item/circuitboard/research_chem_consumer(null)
+
+/obj/machinery/research_chem_consumer/update_icon()
+	if(panel_open)
+		icon_state = "chemical_open"
+	else if(stat & NOPOWER)
+		icon_state = "chemical_powerless"
+	else if(istype(beaker))
+		icon_state = "chemical_on"
+	else
+		icon_state = "chemical_idle"
+
+/obj/machinery/research_chem_consumer/power_change()
+	..()
+	update_icon()
+
+/obj/machinery/research_chem_consumer/screwdriver_act(mob/user, obj/item/I)
+	. = TRUE
+	default_deconstruction_screwdriver(user, icon_state, icon_state, I)
+	update_icon()
+
+/obj/machinery/research_chem_consumer/crowbar_act(mob/user, obj/item/I)
+	. = TRUE
+	default_deconstruction_crowbar(user, I)
+
+/obj/machinery/research_chem_consumer/attackby(obj/item/I, mob/user, params)
+	if(isrobot(user))
+		return
+
+	if(beaker)
+		to_chat(user, "<span class='warning'>Something is already loaded into the machine.</span>")
+		return
+
+	if(istype(I, /obj/item/reagent_containers/glass) || istype(I, /obj/item/reagent_containers/food/drinks))
+		if(panel_open)
+			to_chat(user, "<span class='notice'>Close the maintenance panel first.</span>")
+			return
+		if(!user.drop_item())
+			to_chat(user, "<span class='warning'>[I] is stuck to you!</span>")
+			return
+		beaker =  I
+		I.forceMove(src)
+		to_chat(user, "<span class='notice'>You insert [I] into [src].</span>")
+		SStgui.update_uis(src) // update all UIs attached to src
+		update_icon()
+		return
+
+	return ..()
+
+/obj/machinery/research_chem_consumer/attack_hand(mob/user)
+	ui_interact(user)
+
+/obj/machinery/research_chem_consumer/proc/eject_beaker()
+	if(beaker)
+		beaker.forceMove(get_turf(src))
+		if(Adjacent(usr) && !issilicon(usr))
+			usr.put_in_hands(beaker)
+		beaker = null
+		update_icon()
+
+/obj/machinery/research_chem_consumer/deconstruct()
+	if(beaker)
+		beaker.forceMove(get_turf(src))
+		beaker = null
+	return ..()
+
+/obj/machinery/research_chem_consumer/AltClick(mob/user)
+	if(!ishuman(usr))
+		return
+	if(!istype(user) || user.incapacitated())
+		to_chat(user, "<span class='warning'>You can't do that right now!</span>")
+		return
+	if(!in_range(src, user))
```

Comment:
`Adjacent()` instead, maybe?
```suggestion
	if(!Adjacent(user))
```

Reply:
`Adjacent` can recurse, so this is better